backend/app/modules/user/allocations/service/pending.py

```diff
@@ -125,10 +138,17 @@ def simulate_allocation(
     ) -> Tuple[float, int, List[ProjectRewardDTO]]:
         projects_settings = context.epoch_settings.project
         projects = context.projects_details.projects
+        epoch_num = context.epoch_details.epoch_num
+
         matched_rewards = self.octant_rewards.get_matched_rewards(context)
-        all_allocations_before = database.allocations.get_all(
-            context.epoch_details.epoch_num
+        all_allocations_before = database.allocations.get_all_with_uqs(epoch_num)
+        user_uq = database.uniqueness_quotient.get_uq_by_address(
+            user_address, epoch_num
         )
+        if user_uq is not None:
```

Comment:
I think this is related to the first comment - uq is `None` and the simulation will break. I can see two options here:
- we call `calculate_uq` and provide a real value of it to the simulation at that point in time. Trade off is that it might be slow.
- we assume the value - 0.2 or 1, but then result of the allocation might differ from the simulation

I opt for the first option

Reply:
you're right, I've changed it to calculate_uq dynamically when not provided for the simulation logic; i was wondering if it applies to the EstimatedProjectsRewards too but I figured out it assumes the allocations are already in the database so there's no need to do such dynamic computing